Tests/MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -64,7 +64,11 @@ class NavigationServiceTests: TestCase {
     }
 
     func testDefaultUserInterfaceUsage() {
-        XCTAssertTrue(!dependencies.navigationService.eventsManager.usesDefaultUserInterface, "MapboxCoreNavigationTests shouldn't have an implicit dependency on MapboxNavigation due to removing the Example application target as the test host.")
+        let usesDefaultUserInterface = dependencies.navigationService.eventsManager.usesDefaultUserInterface
+        // Even though `MapboxCoreNavigationTests` does not directly link `MapboxNavigation`, it uses
+        // `TestHelper`, which in turn uses `MapboxNavigation`. This means that its bundle will be present
+        // in `MapboxCoreNavigationTests.xctest`.
+        XCTAssertTrue(usesDefaultUserInterface, "Due to indirect linkage, `MapboxNavigation` will be linked to `MapboxCoreNavigationTests`.")
```

Comment:
This looks like an important change. Previously we incorrectly reported `mapboxNavigationIfInstalled`, because `Bundle.allBundles` did not return all used bundles. As I can see `usesDefaultUserInterface` is used in multiple places while sending navigation events, so it'd be good to understand consequences. 

/cc @mapbox/navigation-ios